src/main/java/org/traccar/protocol/LaipacProtocolDecoder.java

```diff
@@ -158,10 +169,44 @@ private void sendAcknowledge(
 
     }
 
+    protected Object decodeEavsys(
+            String sentence, Channel channel, SocketAddress remoteAddress) {
+
+        Parser parser = new Parser(PATTERN_EAVSYS, sentence);
+        if (!parser.matches()) {
+            return null;
+        }
+
+        DeviceSession deviceSession =
+            getDeviceSession(channel, remoteAddress, parser.next());
+        if (deviceSession == null) {
+            return null;
+        }
+
+        Position position = new Position(getProtocolName());
+        position.setDeviceId(deviceSession.getDeviceId());
+
+        getLastLocation(position, null);
+
+        position.set(Position.KEY_ICCID, parser.next());
+
+        String phoneNumber = parser.next();
+        if (!phoneNumber.isEmpty()) {
```

Comment:
Instead of checks like this you should change regex to only capture non-empty values. For example:
```java
expression("(\\+?[0-9]+)?,")
```

Reply:
You're right, less code, more efficient, null if not found. Great!